### PR TITLE
dvc 2.5.4

### PR DESCRIPTION
**Pull request by efiop**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by efiop** [comment]
Looks like pillow is failing now:

```
2021-07-15T22:56:00.7712960Z     The headers or library files could not be found for jpeg,
2021-07-15T22:56:00.7713470Z     a required dependency when compiling Pillow from source.
2021-07-15T22:56:00.7713810Z 
2021-07-15T22:56:00.7714120Z     Please see the install instructions at:
2021-07-15T22:56:00.7714720Z        https://pillow.readthedocs.io/en/latest/installation.html
2021-07-15T22:56:00.7715220Z 
2021-07-15T22:56:00.7715500Z     Traceback (most recent call last):
2021-07-15T22:56:00.7716180Z       File "/private/tmp/pip-req-build-rq5gipxc/setup.py", line 976, in <module>
2021-07-15T22:56:00.7716680Z         setup(
2021-07-15T22:56:00.7717400Z       File "/opt/homebrew/Cellar/dvc/2.5.4/libexec/lib/python3.9/site-packages/setuptools/__init__.py", line 153, in setup
2021-07-15T22:56:00.7718070Z         return distutils.core.setup(**attrs)
2021-07-15T22:56:00.7718790Z       File "/opt/homebrew/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/lib/python3.9/distutils/core.py", line 148, in setup
2021-07-15T22:56:00.7719440Z         dist.run_commands()
2021-07-15T22:56:00.7720060Z       File "/opt/homebrew/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/lib/python3.9/distutils/dist.py", line 966, in run_commands
2021-07-15T22:56:00.7720740Z         self.run_command(cmd)
2021-07-15T22:56:00.7721370Z       File "/opt/homebrew/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/lib/python3.9/distutils/dist.py", line 985, in run_command
2021-07-15T22:56:00.7722010Z         cmd_obj.run()
2021-07-15T22:56:00.7722750Z       File "/opt/homebrew/Cellar/dvc/2.5.4/libexec/lib/python3.9/site-packages/setuptools/command/install.py", line 61, in run
2021-07-15T22:56:00.7723410Z         return orig.install.run(self)
2021-07-15T22:56:00.7724100Z       File "/opt/homebrew/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/lib/python3.9/distutils/command/install.py", line 546, in run
2021-07-15T22:56:00.7724930Z         self.run_command('build')
2021-07-15T22:56:00.7725550Z       File "/opt/homebrew/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/lib/python3.9/distutils/cmd.py", line 313, in run_command
2021-07-15T22:56:00.7726310Z         self.distribution.run_command(command)
2021-07-15T22:56:00.7727040Z       File "/opt/homebrew/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/lib/python3.9/distutils/dist.py", line 985, in run_command
2021-07-15T22:56:00.7727680Z         cmd_obj.run()
2021-07-15T22:56:00.7728280Z       File "/opt/homebrew/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/lib/python3.9/distutils/command/build.py", line 135, in run
2021-07-15T22:56:00.7728960Z         self.run_command(cmd_name)
2021-07-15T22:56:00.7729590Z       File "/opt/homebrew/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/lib/python3.9/distutils/cmd.py", line 313, in run_command
2021-07-15T22:56:00.7730330Z         self.distribution.run_command(command)
2021-07-15T22:56:00.7731070Z       File "/opt/homebrew/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/lib/python3.9/distutils/dist.py", line 985, in run_command
2021-07-15T22:56:00.7731770Z         cmd_obj.run()
2021-07-15T22:56:00.7732500Z       File "/opt/homebrew/Cellar/dvc/2.5.4/libexec/lib/python3.9/site-packages/setuptools/command/build_ext.py", line 79, in run
2021-07-15T22:56:00.7733100Z         _build_ext.run(self)
2021-07-15T22:56:00.7733730Z       File "/opt/homebrew/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/lib/python3.9/distutils/command/build_ext.py", line 340, in run
2021-07-15T22:56:00.7734430Z         self.build_extensions()
2021-07-15T22:56:00.7735150Z       File "/private/tmp/pip-req-build-rq5gipxc/setup.py", line 788, in build_extensions
2021-07-15T22:56:00.7735850Z         raise RequiredDependencyException(f)
2021-07-15T22:56:00.7736500Z     __main__.RequiredDependencyException: jpeg
2021-07-15T22:56:00.7736910Z 
2021-07-15T22:56:00.7737300Z     During handling of the above exception, another exception occurred:
2021-07-15T22:56:00.7737670Z 
2021-07-15T22:56:00.7737950Z     Traceback (most recent call last):
2021-07-15T22:56:00.7738320Z       File "<string>", line 1, in <module>
2021-07-15T22:56:00.7738990Z       File "/private/tmp/pip-req-build-rq5gipxc/setup.py", line 1034, in <module>
2021-07-15T22:56:00.7739660Z         raise RequiredDependencyException(msg)
2021-07-15T22:56:00.7740310Z     __main__.RequiredDependencyException:
2021-07-15T22:56:00.7740700Z 
2021-07-15T22:56:00.7741040Z     The headers or library files could not be found for jpeg,
2021-07-15T22:56:00.7741550Z     a required dependency when compiling Pillow from source.
2021-07-15T22:56:00.7741880Z 
2021-07-15T22:56:00.7742240Z     Please see the install instructions at:
2021-07-15T22:56:00.7742870Z        https://pillow.readthedocs.io/en/latest/installation.html
2021-07-15T22:56:00.7743360Z 
2021-07-15T22:56:00.7743500Z 
2021-07-15T22:56:00.7744030Z     Running setup.py install for Pillow: finished with status 'error'
```

**Comment by nandahkrishna** [comment]
Thanks for working on this. Could you update this to use `networkx` 2.5.1 instead? As noted in the previous attempt (#80829, specifically https://github.com/Homebrew/homebrew-core/pull/80829#issuecomment-877834721), there are issues due to the default dependencies added in the new version of `networkx`.

By using 2.5.1 and removing all the additional dependencies brought in (like `scipy`, `matplotlib`, `pandas`, etc.) we could get this to build.

**Comment by cho-m** [comment]
@nandahkrishna On side note, it looks like `pipgrip` is unable to detect yanked releases, which means Homebrew's `update-python-resources`/`bump-formula-pr` logic may potentially pull in unstable versions of pypi packages.

**Comment by nandahkrishna** [comment]
Thanks, @cho-m. We may want to open an issue at Homebrew/brew or even directly with `pipgrip` to improve this behaviour. I'll try to take a look at this some time today.

File: Formula/dvc.rb [hunk 1]
```diff
@@ -139,15 +140,20 @@ class Dvc < Formula
...
   resource "chardet" do
     url "https://files.pythonhosted.org/packages/ee/2d/9cdc2b527e127b4c9db64b86647d567985940ac3698eeabc7ffaccb4ea61/chardet-4.0.0.tar.gz"
     sha256 "0d6f53a15db4120f2b08c94f11e7d93d2c911ee118b6b30a04ec3ee8310179fa"
   end
 
+  resource "charset-normalizer" do
+    url "https://files.pythonhosted.org/packages/e1/e6/707ff774274c66f1db739ff4d7eb493252899ed86f0018ac3e9e462fe3de/charset-normalizer-2.0.2.tar.gz"
+    sha256 "951567c2f7433a70ab63f1be67e5ee05d3925d9423306ecb71a3b272757bcc95"
+  end
+
```
Comment (efiop) [inline Formula/dvc.rb:156]:
```suggestion
  resource "charset-normalizer" do
    url "https://files.pythonhosted.org/packages/e1/e6/707ff774274c66f1db739ff4d7eb493252899ed86f0018ac3e9e462fe3de/charset-normalizer-2.0.2.tar.gz"
    sha256 "951567c2f7433a70ab63f1be67e5ee05d3925d9423306ecb71a3b272757bcc95"
  end

```

File: Formula/dvc.rb [hunk 2]
```diff
@@ -173,9 +179,14 @@ class Dvc < Formula
     sha256 "3d10de8116d25649631977cb37da6cbdd2d6fa0e0281d014a5b7d337255ca713"
   end
 
+  resource "Cycler" do
+    url "https://files.pythonhosted.org/packages/c2/4b/137dea450d6e1e3d474e1d873cd1d4f7d3beed7e0dc973b06e8e10d32488/cycler-0.10.0.tar.gz"
+    sha256 "cd7b2d1018258d7247a71425e9f26463dfb444d411c39569972f4ce586b0c9d8"
+  end
+
```
Comment (efiop) [inline Formula/dvc.rb:181]:
```suggestion
```

File: Formula/dvc.rb [hunk 3]
```diff
@@ -324,8 +325,13 @@ class Dvc < Formula
...
   resource "jsonpath-ng" do
-    url "https://files.pythonhosted.org/packages/c5/d0/c4b2fa7e00e69670a92b103761b4e10a4bdaca109818d44753219c20b7be/jsonpath-ng-1.5.2.tar.gz"
-    sha256 "144d91379be14d9019f51973bd647719c877bfc07dc6f3f5068895765950c69d"
+    url "https://files.pythonhosted.org/packages/cd/c2/5cd85d8a7de7818892fd8f751d30b6d050a8471c6647d2fda47cc542f9b4/jsonpath-ng-1.5.3.tar.gz"
+    sha256 "a273b182a82c1256daab86a313b937059261b5c5f8c4fa3fc38b882b344dd567"
+  end
+
+  resource "kiwisolver" do
+    url "https://files.pythonhosted.org/packages/90/55/399ab9f2e171047d28933ae4b686d9382d17e6c09a01bead4a6f6b5038f4/kiwisolver-1.3.1.tar.gz"
+    sha256 "950a199911a8d94683a6b10321f9345d5a3a8433ec58b217ace979e18f16e248"
```
Comment (efiop) [inline Formula/dvc.rb:334]:
```suggestion
```

File: Formula/dvc.rb [hunk 4]
```diff
@@ -338,6 +340,11 @@ class Dvc < Formula
     sha256 "571f295c2534e20ab096915681ed57e851e2ea18e7c34bde1f5304e175ae68cf"
   end
 
+  resource "matplotlib" do
+    url "https://files.pythonhosted.org/packages/60/d3/286925802edaeb0b8834425ad97c9564ff679eb4208a184533969aa5fc29/matplotlib-3.4.2.tar.gz"
+    sha256 "d8d994cefdff9aaba45166eb3de4f5211adb4accac85cbf97137e98f26ea0219"
+  end
+
```
Comment (efiop) [inline Formula/dvc.rb:347]:
```suggestion
```

File: Formula/dvc.rb [hunk 5]
```diff
@@ -388,6 +395,11 @@ class Dvc < Formula
     sha256 "7dc96269f53a4ccec5c0670940a4281106dd0bb343f47b7471f779df49c2fbe7"
   end
 
+  resource "pandas" do
+    url "https://files.pythonhosted.org/packages/53/05/bf382e8bc60731906a2e7261648bcea3a6b309ad2b9952010737a1b9413e/pandas-1.3.0.tar.gz"
+    sha256 "c554e6c9cf2d5ea1aba5979cc837b3649539ced0e18ece186f055450c86622e2"
+  end
+
```
Comment (efiop) [inline Formula/dvc.rb:402]:
```suggestion
```

File: Formula/dvc.rb [hunk 6]
```diff
@@ -549,13 +561,18 @@ class Dvc < Formula
...
   resource "s3transfer" do
-    url "https://files.pythonhosted.org/packages/66/f5/5ca537483fa5e96fbd455f52a69fc70c5f659f7e8c9189a1dbc211e1ccf9/s3transfer-0.3.7.tar.gz"
-    sha256 "35627b86af8ff97e7ac27975fe0a98a312814b46c6333d8a6b889627bcd80994"
+    url "https://files.pythonhosted.org/packages/27/90/f467e516a845cf378d85f0a51913c642e31e2570eb64b352c4dc4c6cbfc7/s3transfer-0.4.2.tar.gz"
+    sha256 "cb022f4b16551edebbb31a377d3f09600dbada7363d8c5db7976e7f47732e1b2"
+  end
+
+  resource "scipy" do
+    url "https://files.pythonhosted.org/packages/bb/bb/944f559d554df6c9adf037aa9fc982a9706ee0e96c0d5beac701cb158900/scipy-1.7.0.tar.gz"
+    sha256 "998c5e6ea649489302de2c0bc026ed34284f531df89d2bdc8df3a0d44d165739"
```
Comment (efiop) [inline Formula/dvc.rb:575]:
```suggestion
```

File: Formula/dvc.rb [hunk 7]
```diff
@@ -324,8 +325,9 @@ class Dvc < Formula
   end
 
   resource "jsonpath-ng" do
-    url "https://files.pythonhosted.org/packages/c5/d0/c4b2fa7e00e69670a92b103761b4e10a4bdaca109818d44753219c20b7be/jsonpath-ng-1.5.2.tar.gz"
-    sha256 "144d91379be14d9019f51973bd647719c877bfc07dc6f3f5068895765950c69d"
+    url "https://files.pythonhosted.org/packages/cd/c2/5cd85d8a7de7818892fd8f751d30b6d050a8471c6647d2fda47cc542f9b4/jsonpath-ng-1.5.3.tar.gz"
+    sha256 "a273b182a82c1256daab86a313b937059261b5c5f8c4fa3fc38b882b344dd567"
+  end
```
Comment (efiop) [inline Formula/dvc.rb:330]:
```suggestion
```

File: Formula/dvc.rb [hunk 8]
```diff
@@ -661,6 +661,13 @@ def install
...
     output = shell_output("#{bin}/dvc doctor 2>&1")
-    assert_match "Supports: azure, gdrive, gs, http, https, s3, ssh, oss, webdav, webdavs", output
+    assert_match "azure", output
+    assert_match "gdrive", output
+    assert_match "gs", output
+    assert_match "http", output
+    assert_match "s3", output
+    assert_match "ssh", output
+    assert_match "oss", output
+    assert_match "webdav", output
```
Comment (efiop) [inline Formula/dvc.rb:671]: For the record: the format of `dvc doctor` has changed and old test no longer worked, so I've adjusted it to be simpler.

**Comment by efiop** [comment]
@nandahkrishna @chenrui333 Thanks for the help! Adjusted to use an older networkx for now, so ready for review. 🙂  Once a new networkx version is released, I'll update a following dvc release accordingly as well.

**Review by SMillerDev — APPROVED** [review]

**Comment by nandahkrishna** [comment]
Thanks, @efiop. If the next NetworkX release removes the default dependencies, just `bump-formula-pr` should be fine without any additional changes.

**Pull request closed without merging**